nova/virt/vmwareapi/cluster_util.py

```diff
@@ -429,3 +430,26 @@ def update_cluster_das_vm_override(session, cluster, vm_ref, operation='add',
     config_spec.dasVmConfigSpec = [das_vm_spec]
 
     reconfigure_cluster(session, cluster, config_spec)
+
+
+def get_cluster_median_cpu_mhz(session, cluster) -> int:
+    """Get the median CPU frequency for all hosts in a cluster.
+
+    We need this for reservation, since flavors specify reservation in cores,
+    but reservation of a VMware VM's CPUs is specified in Hz. We calculate the
+    median to prevent outliers (esp. hosts that might have much _less_ CPUs for
+    some reason) from skewing the result.
+    """
+    hosts = session._call_method(vutil, "get_object_property", cluster,
+                                 "host")
+    if not hosts:
+        raise CpuReservationNoHostsError(cluster=cluster)
+    host_hz = [
+        session._call_method(vutil, "get_object_property", host,
+                             "hardware.cpuInfo.hz")
+        for host in hosts.ManagedObjectReference]
+    host_hz.sort()
+    # NOTE(jakobk): Yes, the proper median for an even number of values is the
+    # average of the two middle values, but we don't need that precision.
+    # And yes, it's SI MHz, not MiHz.
```

Comment:
Can we drop the `Yes` and `Any yes`? Sounds off to me.

Reply:
obsoleted